### PR TITLE
Add support for building for Ledger Nano S Plus

### PR DESCRIPTION
**Pull request by orhoj**
## Purpose
Adding support for building the application for the Ledger Nano S Plus device.

## Changes
- Added dependency to the Nano S Plus SDK.
- Updated Dockerfile to use the compatible ledger-app-builder. Note that a different `clang` version is to be used now, otherwise building will fail. Check the README, which leads to the ledger-app-builder setup as it shows what dependencies to use for building locally.
- Updated release build to also build in the name of the target device (nanos vs. nanos-plus).
- Update Zemu dependency to latest available version.
- Added end to end tests for the Nano S Plus.
- The version is not bumped, as no changes to the actual application were required.

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

**Comment by shjortConcordium** [comment]
Doesn't work with the desktop wallet.

File: tests/src/configureBaker.test.ts [hunk 1]
```diff
@@ -224,43 +254,51 @@ test('[NANO S] Configure-baker: big URL only', setupZemu('nanos', async (sim, tr
...
 
 test('[NANO S] Configure-baker: Single commission rate', setupZemu('nanos', async (sim, transport) => {
     const bitmap = '0040';
     await configureBakerStep0(bitmap, transport);
-    await configureBakerCommissionStep(false, true, false, true, 'nanos', "single_commission_rate" ,"9b5f30ce60f17c137de0ff1135ebd899b9ee00418928ee2f4f986c651508637c8bad35487518e6e2284a1b17b28358a38c33a9c015f78e33c66e79dcf471160b9000", sim, transport);
+    await configureBakerCommissionStep(false, true, false, true, 'nanos', 'single_commission_rate', '9b5f30ce60f17c137de0ff1135ebd899b9ee00418928ee2f4f986c651508637c8bad35487518e6e2284a1b17b28358a38c33a9c015f78e33c66e79dcf471160b9000', sim, transport);
 }));
 
-test('[NANO X] Configure-baker: Single commission rate', setupZemu('nanox', async (sim, transport) => {
+async function configureBakerSingleCommissionRateXAndSP(sim: Zemu, transport: Transport, device: LedgerModel) {
```
Comment (shjortConcordium) [inline tests/src/configureBaker.test.ts:293]: Not very important, but for some of these tests, we could actually use this function for NANO S also?

File: tests/src/simpleTransfer.test.ts [hunk 2]
```diff
@@ -1,10 +1,11 @@
+import Transport from '@ledgerhq/hw-transport';
+import Zemu from '@zondax/zemu';
 import { setupZemu } from './options';
 
 test('[NANO S] Sign a valid simple transfer with memo', setupZemu('nanos', async (sim, transport) => {
     let data = Buffer.from('08000004510000000000000000000000000000000000000002000000000000000020a845815bd43a1999e90fbf971537a70392eb38f89e6bd32b3dd70e1a9551d7000000000000000a0000000000000064000000290000000063de5da71620a845815bd43a1999e90fbf971537a70392eb38f89e6bd32b3dd70e1a9551d70005', 'hex');
     transport.send(0xe0, 0x32, 0x01, 0x00, data);
 
-    console.log('Go for it');
```
Comment (shjortConcordium) [inline tests/src/simpleTransfer.test.ts:7]: :cry:

**Comment by orhoj** [comment]
> Doesn't work with the desktop wallet.

It will with the delivery of https://github.com/Concordium/concordium-desktop-wallet/pull/321

**Review by shjortConcordium — APPROVED** [review]

**Pull request merged**